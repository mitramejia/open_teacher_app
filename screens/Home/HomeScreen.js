import React from 'react';
import { Button, Text, Content } from 'native-base';
import StudentHomeScreen from '../Student/Home/StudentHomeScreen';
import LoginScreen from '../Login/LoginScreen';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userIsLoggedIn: false,
    };
  }

  static navigationOptions = {
    header: null,
  };

  render() {
    if (this.state.userIsLoggedIn) {
      return <StudentHomeScreen navigation={this.props.navigation} />;
    } else {
      return <LoginScreen navigation={this.props.navigation} />;
    }
  }
}
