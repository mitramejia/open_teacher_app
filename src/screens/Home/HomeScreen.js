import React from 'react';
import { AsyncStorage } from 'react-native';
import { Button, Content } from 'native-base';
import PropTypes from 'prop-types';
import SearchTutorScreen from '../Student/SearchTutor/SearchTutorScreen';
import LoginScreen from '../Login/LoginScreen';
import TutorHomeScreen from '../Tutor/Home/TutorHomeScreen';

export class HomeScreen extends React.Component {
  state = {
    user: null,
  };

  static navigationOptions = {
    header: null,
  };

  _logout = async () => {
    await AsyncStorage.removeItem('token');
    this.props.client.resetStore();
  };

  render() {
    if (this.props.data.user) {
      if (this.props.data.user.isTutor) {
        return <TutorHomeScreen user={this.props.data.user} navigation={this.props.navigation} />;
      } else {
        return <SearchTutorScreen user={this.props.data.user} navigation={this.props.navigation} />;
      }
    } else {
      return <LoginScreen client={this.props.client} navigation={this.props.navigation} />;
    }
  }
}

HomeScreen.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.any.isRequired,
  }),
  navigation: PropTypes.object.isRequired,
};

export default HomeScreen;
