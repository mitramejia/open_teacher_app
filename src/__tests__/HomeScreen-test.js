import React from 'react';
import HomeScreen from '../screens/Home/HomeScreen';
import renderer from 'react-test-renderer';

it('renders the loading screen', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the root without loading screen', () => {
  const tree = renderer.create(<HomeScreen skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
