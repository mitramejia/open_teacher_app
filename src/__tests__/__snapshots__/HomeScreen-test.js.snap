// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading screen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  styles={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      styles={undefined}
    >
      <Image
        source={1}
        styles={
          Object {
            "alignContent": "center",
            "height": 280,
            "marginTop": 3,
            "resizeMode": "contain",
            "width": 200,
          }
        }
      />
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        styles={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#1D374C",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 26,
            "paddingRight": 26,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          styles={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Inicia Sesión con Facebook
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        styles={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#007aff",
            "borderLeftWidth": 1,
            "borderRadius": 57,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "borderWidth": 1,
            "elevation": null,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 26,
            "paddingRight": 26,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          styles={
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Crear una Cuenta
        </Text>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      styles={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#1D374C",
            "marginTop": 40,
            "textDecorationLine": "underline",
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      Soy un estudiante
    </Text>
  </View>
</RCTScrollView>
`;

exports[`renders the root without loading screen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  styles={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      styles={undefined}
    >
      <Image
        source={1}
        styles={
          Object {
            "alignContent": "center",
            "height": 280,
            "marginTop": 3,
            "resizeMode": "contain",
            "width": 200,
          }
        }
      />
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        styles={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#1D374C",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 26,
            "paddingRight": 26,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          styles={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Inicia Sesión con Facebook
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        styles={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#007aff",
            "borderLeftWidth": 1,
            "borderRadius": 57,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "borderWidth": 1,
            "elevation": null,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 26,
            "paddingRight": 26,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          styles={
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Crear una Cuenta
        </Text>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      styles={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#1D374C",
            "marginTop": 40,
            "textDecorationLine": "underline",
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      Soy un estudiante
    </Text>
  </View>
</RCTScrollView>
`;
